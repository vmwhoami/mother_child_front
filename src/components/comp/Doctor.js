import React from 'react';
import PropTypes from 'prop-types';
import css from '../../css/doctors.module.css';

const Doctor = ({ doctor }) => {
  const { name, title, img } = doctor;
  return (
    <div className={css.doctor}>
      <h2>{name}</h2>
      <p>{title}</p>
      <img src={img} alt="name" />
    </div>
  );
};
Doctor.propTypes = {
  doctor: PropTypes.instanceOf(Object).isRequired,
};
export default Doctor;
