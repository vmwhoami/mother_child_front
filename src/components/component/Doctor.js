import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Doctor = ({ doctor, selectDoc }) => {
  const {
    name, title, img, id, recieving_hours: hours,
  } = doctor;

  return (
    <div className="doctor">
      <div className="docimgContainer">
        <img className="docimg" src={img} alt={name} />
      </div>
      <h4>{name}</h4>
      <span>{title}</span>
      <p>{`Receiving hours: ${hours}`}</p>

      <Link to={`/${id}`} onClick={() => selectDoc(id)}>
        <button className="button" type="button">Book an appointment</button>
      </Link>
    </div>

  );
};
Doctor.propTypes = {
  doctor: PropTypes.instanceOf(Object).isRequired,
  selectDoc: PropTypes.func.isRequired,
};
export default Doctor;
