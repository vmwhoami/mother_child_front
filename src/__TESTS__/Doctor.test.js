import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Doctor from '../components/component/Doctor';

const doctor = {};

const selectDoc = jest.fn();
describe('Doctor', () => {
  test('snapshot renders correctly', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Doctor doctor={doctor} selectDoc={selectDoc} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
