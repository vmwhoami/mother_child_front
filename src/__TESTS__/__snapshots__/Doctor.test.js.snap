// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doctor snapshot renders correctly 1`] = `
<div
  className="doctor"
>
  <div
    className="docimgContainer"
  >
    <img
      className="docimg"
    />
  </div>
  <h4 />
  <span />
  <p>
    Receiving hours: undefined
  </p>
  <a
    href="/undefined"
    onClick={[Function]}
  >
    <button
      className="button"
      type="button"
    >
      Book an appointment
    </button>
  </a>
</div>
`;
